src/ensembl/src/content/app/custom-download/containers/content/result-holder/ResultHolder.tsx

```diff
@@ -20,32 +20,40 @@ import {
 import styles from './ResultHolder.scss';
 
 import { CircleLoader } from 'src/shared/loader/Loader';
+import JSONValue from 'src/shared/types/JSON';
 
 import {
-  getSelectedAttributes,
   formatResults,
+  flattenObject,
   getEndpointUrl
 } from './resultHolderHelper';
 
 type Props = StateProps & DispatchProps;
 
 const ResultHolder = (props: Props) => {
-  const selectedAttributes: AttributesSection = getSelectedAttributes(
-    props.attributes
-  );
-
   useEffect(() => {
-    if (!selectedAttributes.length && props.preview.results) {
+    const flatSelectedAttributes: { [key: string]: boolean } = flattenObject(
+      props.selectedAttributes
+    );
+
+    const totalSelectedAttributes = keys(flatSelectedAttributes).length;
+    if (!totalSelectedAttributes && props.preview.results) {
       props.clearPreviewResult({});
```

Comment:
Why does a function called `clearPreviewResult` accept an argument?

If it were declared in mapDispatchToProps as  `clearPreviewResult: () => setPreviewResult.success({})`, wouldn't it work?

Reply:
That's done